### PR TITLE
python module: implement the new preserve_path kwarg for install_sources too

### PR DESCRIPTION
**Pull request by eli-schwartz**
Based on the new feature in install_headers.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
I'm actually wondering now if it makes more sense to add this to install_data and let python inherit it, actually...

**Comment by rgommers** [comment]
This would be great to have, thanks @eli-schwartz! It's indeed much better than one of the two alternatives:

1. Use `subdir()` followed by a separate `meson.build` file in each sub-directory.
2. Use multiple separate `install_sources` invocations. Here's an example from SciPy:

```meson
py3.install_sources(
  python_sources,  # a set of .py files
  pure: false,
  subdir: 'scipy/ndimage/tests'
)

py3.install_sources([
    'data/label_inputs.txt',
    'data/label_results.txt',
    'data/label_strels.txt',
  ],
  pure: false,
  subdir: 'scipy/ndimage/tests/data'
)
```

> I'm actually wondering now if it makes more sense to add this to `install_data` and let python inherit it, actually...

I guess that makes sense, but I don't use `install_data` so I can't say more.

**Comment by eli-schwartz** [comment]
I ended up deciding to apply it to install_data too. IMHO it can be useful there as well, also, it has a better doc setup. :p

@rgommers



> Also, now that you're working on making `python` easier to use, another one that comes to mind is not having to specify:
> 
> ```meson
> install: true  # in both install_sources and extension_module
> ```

You don't need to specify install, specifically, for install_sources at all, only for extension_module?

> If this makes sense, I'm happy to open a separate issue with a more detailed description.

I'm not completely sure what to think of this, but definitely open a new ticket for it. :)

File: docs/yaml/functions/install_data.yaml [hunk 1]
```diff
@@ -43,6 +43,15 @@ kwargs:
       to install only a subset of the files. By default these files have no install
       tag which means they are not being installed when `--tags` argument is specified.
 
+  preserve_path:
+    type: bool
+    since: 0.63.0
```
Comment (dcbaker) [inline docs/yaml/functions/install_data.yaml:48]: neds an update, obviously :)

**Review by dcbaker — APPROVED** [review]
With the versions updated, this LGTM

File: mesonbuild/interpreter/type_checking.py [hunk 2]
```diff
@@ -438,3 +438,5 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
     convertor=variables_convertor,
     default={},
 )
+
+PRESERVE_PATH_KW: KwargInfo[bool] = KwargInfo('preserve_path', bool, default=False, since='0.63.0')
```
[the remark this answers is not in the document]
Comment (eli-schwartz) [reply mesonbuild/interpreter/type_checking.py:442]: Nah, this one is the base implementation used by install_headers too. :) It gets evolved instead.

**Pull request merged**